src/editors/containers/ProblemEditor/components/SelectTypeModal/hooks.js

```diff
@@ -15,7 +16,11 @@ export const selectHooks = () => {
   };
 };
 
-export const onSelect = ( setProblemType, selected ) => () => {
+export const onSelect = ( setProblemType, selected, updateField ) => () => {
+  if (Object.values(AdvanceProblemKeys).includes(selected)) {
+    console.log(AdvanceProblems[selected]);
+    updateField({rawOLX: AdvanceProblems[selected].template})
```

Comment:
We should just create a new reducer called `setOlxTemplate` or something. Much more readable and direct.

Reply:
I don't think that creating another reducer makes the call more readable and direct. This is the only place the reducer would be used so it seems unnecessary to create a new reducer function for just one call. Unlike `setProblemType` which is probably used in multiple location.

Reply:
Ok, that makes sense to me. I was thinking about it from the point of view of reading an onSelect call and getting an understanding of what it's doing based on the inputs.